src/main/flight/pid.c

```diff
@@ -130,6 +130,91 @@ float getdT (void) {
+            // horizon_incl_fact 100 to 200 => 2700 to 900
+            //  (represents where leveling goes to zero):
+            cutoffDeciDegrees = (250-horizonInclFact) * 18;
+            // inclinationLevelRatio (0 to 100) is smaller (less leveling)
+            //  for larger inclinations; 0 at cutoffDeciDegrees value:
+            inclinationLevelRatio = constrain(
+                              (((cutoffDeciDegrees-currentInclination)*10) /
+                                           (cutoffDeciDegrees/10)), 0, 100);
+            // apply configured horizon sensitivity:
+            if (horizonSensit <= 0) {       // zero means no leveling
+                horizonLevelStrength = 0;
+            } else {
+                // when stick is near center (horizonLevelStrength ~= 100)
+                //  H_sensitivity value has little effect,
+                // when stick is deflected (horizonLevelStrength near 0)
+                //  H_sensitivity value has more effect:
+                horizonLevelStrength = constrain(
+                                        (((horizonLevelStrength-100) * 100 /
+                                             horizonSensit) + 100), 0, 100);
+            }
+            // apply inclination ratio, which may lower leveling
+            //  to zero regardless of stick position:
+            horizonLevelStrength =
+                         horizonLevelStrength * inclinationLevelRatio / 100;
+        }
+        else
+          horizonLevelStrength = 0;
+    } else {       // horizon_incl_fact < 100
+        // range 1 (leveling always active when sticks centered)
+        if (horizonInclFact > 0) {
+            // ratio of 100 to 0 (larger means more leveling):
+            factorRatio = 100 - horizonInclFact;
+            // inclinationLevelRatio (0 to 100) is smaller (less leveling)
+            //  for larger inclinations, goes to 100 at inclination level:
+            inclinationLevelRatio =
+                  ((1800-currentInclination)/18 * (100-factorRatio)) / 100 +
+                                                                factorRatio;
+            // apply ratio to configured horizon sensitivity:
+            sensitFact = horizonSensit * inclinationLevelRatio / 100;
+        }
+        else   // horizon_incl_fact=0 for "old" functionality
+            sensitFact = horizonSensit;
+
+        if (sensitFact <= 0) {           // zero means no leveling
+            horizonLevelStrength = 0;
+        } else {
+            // when stick is near center (horizonLevelStrength ~= 100)
+            //  sensitFact value has little effect,
+            // when stick is deflected (horizonLevelStrength near 0)
+            //  sensitFact value has more effect:
+            horizonLevelStrength = constrain(
+                    ((horizonLevelStrength - 100) * 100 / sensitFact + 100),
+                                                                    0, 100);
+        }
+    }
+    return horizonLevelStrength;
```

Comment:
Removing the 'contrain()' from line 176 is problematic because if 'horizonSensit' is small the 'horizonLevelStrength' value at that point can become a large negative value, and the value is then multiplied by 'inclinationLevelRatio' (could overflow 16-bit integer).

--ET